jerry-core/ecma/builtin-objects/ecma-builtin-function-prototype.c

```diff
@@ -122,44 +122,51 @@ ecma_builtin_function_prototype_object_apply (ecma_value_t this_arg, /**< this a
         /* 5. */
         const uint32_t length = ecma_number_to_uint32 (length_number);
 
-        /* 6. */
-        JMEM_DEFINE_LOCAL_ARRAY (arguments_list_p, length, ecma_value_t);
-        uint32_t last_index = 0;
-
-        /* 7. */
-        for (uint32_t index = 0;
-             index < length && ecma_is_value_empty (ret_value);
-             index++)
+        if (length >= 65536)
```

Comment:
`if (length > UINT16_MAX)`

Reply:
It is a coincidence that the two numbers are the same, this value can be 1000, 10000, or whatever number. The point is it should have a limit. The normal JS calls has a limit of 255 arguments, but I think this could be bigger.

Reply:
OK, then please create a new define for it. We shouldn't use magic constants. Please give it a name.

Reply:
In which file should we put it? This .c file, or some header?

Reply:
If it is only used in the .c file, then it is enough to put it into that file.